src/Microsoft.AspNetCore.Server.Kestrel.Configuration/KestrelServerOptionsSetup.cs

```diff
@@ -0,0 +1,111 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Security.Cryptography.X509Certificates;
+using Microsoft.AspNetCore.Certificates.Configuration;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Server.Kestrel.Core;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Logging;
+using Microsoft.Extensions.Options;
+using Microsoft.Extensions.Options.Infrastructure;
+
+namespace Microsoft.AspNetCore.Server.Kestrel.Core.Internal
+{
+    internal class KestrelServerConfigureOptions : ConfigureDefaultOptions<KestrelServerOptions>
```

Comment:
Do we want all of these `ConfigureDefaultOptions` public?   Or just Kestrel's since its doing real work unlike the rest which are usually one liners that only call GetSection().Bind which are mostly private.